apis/placement/v1alpha1/override_types.go

```diff
@@ -48,6 +48,21 @@ type ClusterResourceOverrideSpec struct {
 	// Policy defines how to override the selected resources on the target clusters.
 	// +required
 	Policy *OverridePolicy `json:"policy"`
+
+	// Placement defines whether the override is applied to a specific placement or not.
+	// If so, the override will trigger the placement rollout immediately when the rollout strategy type is RollingUpdate.
+	// Otherwise, it will be applied to the next rollout.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="spec.placement is immutable"
```

Comment:
is this here for the backward compatibility sake? How do we handle the case that the ref is nil? I thought it was too computational heavy?

Reply:
It's mentioned in the comment, if the placement is nil, the override will be applied to the next rollout, and it won't trigger the selected placement immediately.